src/compiler/config/outputs/validate-www.ts

```diff
@@ -34,105 +33,151 @@ export const validateWww = (config: d.ValidatedConfig, diagnostics: d.Diagnostic
+): d.ValidatedOutputTargetWww => {
+  const baseUrl = validateBaseUrl(outputTarget.baseUrl);
+  const dir = getAbsolutePath(config, outputTarget.dir || 'www');
+  const appDir = validateAppDir(baseUrl, dir);
+
+  const validatedOutputTarget = {
+    appDir,
+    baseUrl,
+    buildDir: validateBuildDir(outputTarget.buildDir, appDir),
+    copy: outputTarget.copy === undefined ? [] : outputTarget.copy,
+    dir,
+    empty: isBoolean(outputTarget.empty) ? outputTarget.empty : true,
+    indexHtml: validateIndexHtml(outputTarget.indexHtml, appDir),
+    polyfills: isBoolean(outputTarget.polyfills) ? outputTarget.polyfills : true,
+    prerenderConfig: validatePrerenderConfig(outputTarget.prerenderConfig, baseUrl, config, diagnostics),
+    serviceWorker: validateServiceWorker(config, outputTarget.serviceWorker, appDir),
+    type: WWW,
+  } satisfies d.ValidatedOutputTargetWww;
+
+  return validatedOutputTarget;
+};
+
+const validateAppDir = (baseUrl: string, dir: string): string => {
+  const pathname = new URL(baseUrl, 'http://localhost/').pathname;
+  let appDir = join(dir, pathname);
+  if (appDir.endsWith('/') || appDir.endsWith('\\')) {
+    appDir = appDir.substring(0, appDir.length - 1);
   }
 
-  if (!outputTarget.baseUrl.endsWith('/')) {
+  return appDir;
+};
+
+const validateBaseUrl = (baseUrl: string | undefined): string => {
+  if (!isString(baseUrl)) {
+    baseUrl = '/';
+  }
+  if (!baseUrl.endsWith('/')) {
     // Make sure the baseUrl always finish with "/"
-    outputTarget.baseUrl += '/';
+    baseUrl += '/';
   }
 
-  outputTarget.dir = getAbsolutePath(config, outputTarget.dir || 'www');
+  return baseUrl;
+};
 
-  // Fix "dir" to account
-  const pathname = new URL(outputTarget.baseUrl, 'http://localhost/').pathname;
-  outputTarget.appDir = join(outputTarget.dir, pathname);
-  if (outputTarget.appDir.endsWith('/') || outputTarget.appDir.endsWith('\\')) {
-    outputTarget.appDir = outputTarget.appDir.substring(0, outputTarget.appDir.length - 1);
+const validateBuildDir = (buildDir: string | undefined, appDir: string): string => {
+  buildDir = isString(buildDir) ? buildDir : 'build';
+  if (!isAbsolute(buildDir)) {
+    buildDir = join(appDir, buildDir);
   }
 
-  if (!isString(outputTarget.buildDir)) {
-    outputTarget.buildDir = 'build';
-  }
+  return buildDir;
+};
 
-  if (!isAbsolute(outputTarget.buildDir)) {
-    outputTarget.buildDir = join(outputTarget.appDir, outputTarget.buildDir);
+const validateIndexHtml = (indexHtml: string | undefined, appDir: string): string => {
+  if (!isString(indexHtml)) {
+    indexHtml = 'index.html';
   }
-
-  if (!isString(outputTarget.indexHtml)) {
-    outputTarget.indexHtml = 'index.html';
+  if (!isAbsolute(indexHtml)) {
+    indexHtml = join(appDir, indexHtml);
   }
 
-  if (!isAbsolute(outputTarget.indexHtml)) {
-    outputTarget.indexHtml = join(outputTarget.appDir, outputTarget.indexHtml);
+  return indexHtml;
+};
+
+const validatePrerenderConfig = (
```

Comment:
Moved this over from the separate file where it was before. We only use this helper validating this target and we didn't have any tests written specifically for it